src/Nethermind/Nethermind.Synchronization/Peers/ISyncPeerPool.cs

```diff
@@ -13,7 +14,17 @@ namespace Nethermind.Synchronization.Peers
 {
     public interface ISyncPeerPool : IDisposable
     {
-        Task<SyncPeerAllocation> Allocate(IPeerAllocationStrategy peerAllocationStrategy, AllocationContexts allocationContexts, int timeoutMilliseconds = 0);
+        Task<SyncPeerAllocation> Allocate(
+            IPeerAllocationStrategy peerAllocationStrategy,
+            AllocationContexts allocationContexts,
+            int timeoutMilliseconds = 0,
+            CancellationToken? cancellationToken = null);
+
+        Task<T> AllocateAndRun<T>(
```

Comment:
Can we move it to extension method? Doesn't sit well with me on the interface, it should have only one implementation and is more of helper method.

Reply:
Done